File: Challenges/Coding World Cup/2023/Day 31 - ENG vs AUS - Swap for Maximum Performance.cpp
int maxSwap(vector<int> &a)
{
    // Write your code here
    int n = a.size();
    int m = a[1];

    for (int i = 1; i <= n - 2; i++)
        m = max(m, a[i]);

    return max(a[0] + a[n - 1], max(m + a[0], m + a[n - 1]));
}

File: Challenges/Coding World Cup/2023/Day 31 - ENG vs AUS - Swap for Maximum Performance.java
public class Solution {
    static int maxSwap(int[] a) {
        int n = a.length;
        int m = a[1];

        for (int i = 1; i <= n - 2; i++)
            m = Math.max(m, a[i]);

        return Math.max(a[0] + a[n - 1], Math.max(m + a[0], m + a[n - 1]));
    }
}